sdk/python/feast/pyspark/historical_feature_retrieval_job.py

```diff
@@ -1,19 +1,433 @@
+import abc
+import argparse
 import json
-from datetime import timedelta
-from typing import Any, Dict, List
+from datetime import datetime, timedelta
+from typing import Dict, List, NamedTuple, Optional
 
-from pyspark import SparkFiles
 from pyspark.sql import DataFrame, SparkSession, Window
 from pyspark.sql.functions import col, expr, monotonically_increasing_id, row_number
 
 
+class Source(abc.ABC):
+    """
+    Source for an entity or feature dataframe.
+
+    Attributes:
+        timestamp_column (str): Column representing the event timestamp.
```

Comment:
can we make this `event_timestamp_column`?

Reply:
I prefer `event_timestamp_column` as well, though there are other instances in Feast SDK where `timestamp_column` is used instead (Datasource, BatchSource, for example).

I am happy to make the change though. Should i change it only for `historical_feature_retrieval_job`?

Reply:
Lets just make the change here please.